src/treeview/element.js

```diff
@@ -254,27 +358,188 @@ export default class Element extends Node {
 			return true;
 		}
 
-		// Check name and attributes.
+		// Check element name.
 		if ( this.name != otherElement.name ) {
 			return false;
 		}
 
-		const thisNodeAttrKeys = this.getAttributeKeys();
-		const otherNodeAttrKeys = otherElement.getAttributeKeys();
-		let count = 0;
+		// Check number of attributes, classes and styles.
+		if ( this._attrs.size !== otherElement._attrs.size || this._classes.size !== otherElement._classes.size ||
+			this._styles.size !== otherElement._styles.size ) {
+			return false;
+		}
 
-		for ( let key of thisNodeAttrKeys ) {
-			if ( this.getAttribute( key ) !== otherElement.getAttribute( key ) ) {
+		// Check if attributes are the same.
+		for ( let [ key, value ] of this._attrs ) {
+			if ( !otherElement._attrs.has( key ) || otherElement._attrs.get( key ) !== value ) {
 				return false;
 			}
+		}
 
-			count++;
+		// Check if classes are the same.
+		for ( let className of this._classes ) {
+			if ( !otherElement._classes.has( className ) ) {
+				return false;
+			}
 		}
 
-		if ( count != utils.count( otherNodeAttrKeys ) ) {
-			return false;
+		// Check if styles are the same.
+		for ( let [ property, value ] of this._styles ) {
+			if ( !otherElement._styles.has( property ) || otherElement._styles.get( property ) !== value ) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Adds specified class.
+	 *
+	 * @example
+	 * element.addClass( 'foo' ); // Adds 'foo' class.
+	 * element.addClass( 'foo', 'bar' ); // Adds 'foo' and 'bar' classes.
+	 *
+	 * @param {...String} className
+	 * @fires core.treeView.Node#change
+	 */
+	addClass( ...className ) {
+		this._fireChange( 'ATTRIBUTES', this );
+		className.forEach( name => this._classes.add( name ) );
+	}
+
+	/**
+	 * Removes specified class.
+	 *
+	 * @example
+ 	 * element.removeClass( 'foo' );  // Removes 'foo' class.
+	 * element.removeClass( 'foo', 'bar' ); // Removes both 'foo' and 'bar' classes.
+	 *
+	 * @param {...String} className
+	 * @fires core.treeView.Node#change
+	 */
+	removeClass( ...className ) {
+		this._fireChange( 'ATTRIBUTES', this );
+		className.forEach( name => this._classes.delete( name ) );
+	}
+
+	/**
+	 * Returns true if class is present.
+	 * If more then one class is provided - returns true only when all classes are present.
+	 *
+	 * @example
+	 * element.hasClass( 'foo' ); // Returns true if 'foo' class is present.
+	 * element.hasClass( 'foo', 'bar' ); // Returns true if 'foo' and 'bar' classes are both present.
+	 *
+	 * @param {...String} className
+	 */
+	hasClass( ...className ) {
+		for ( let name of className ) {
+			if ( !this._classes.has( name ) ) {
+				return false;
+			}
 		}
 
 		return true;
 	}
+
+	/**
+	 * Adds style to the element.
+	 *
+	 * @example
+	 * element.setStyle( 'color', 'red' );
+	 * element.setStyle( {
+	 * 		color: 'red',
+	 * 		position: 'fixed'
+	 * } );
+	 *
+	 * @param {String|Object} property Property name or object with property - value pairs.
```

Comment:
It is usually `key - value` or `name - value`, so I would rename property to `key` or `name`.